src/classes/AFFL_AccChange_TDTM.cls

```diff
@@ -56,93 +61,163 @@ public with sharing class AFFL_AccChange_TDTM extends TDTM_Runnable {
-                   		if(affl.Primary__c && affl.Account__c != null && afflOld.Account__c != null
-                       		&& affl.Account__c != afflOld.Account__c && lookupFieldAPIName != null) {
-
-                       		if(affl.Affiliation_Type__c == afflOld.Affiliation_Type__c) {
-                           		//If the primary affl now points to an Account of the same type, update the
-                           		//same primary field in Contact.
-                           		relatedContact.put(lookupFieldAPIName, affl.Account__c);
-                       		} else {
-                           		//If the primary affl now points to an Account of different type, clear the old
-                           		//primary field in Contact and populate the matching new field.
-                           		if (lookupFieldAPINameOld != null) {
-                               		relatedContact.put(lookupFieldAPINameOld, null);
-                               		relatedContact.put(lookupFieldAPIName, affl.Account__c);
-                           		}
-                       		}
-                       		dmlWrapper.objectsToUpdate.add(relatedContact);
-
-                       	}
+                            lookupFieldAPINameOld = AFFL_MultiRecordType_TDTM.afflMapper.contactLabelNames.get(lookupFieldLabelOrAPIOld);
+                    }
+
+                    if (isAfflTypeEnforced == true) {
+                        ERR_ExceptionHandler.handleAfflNullRecordTypeException(affl, afflMapper.validAccRecordTypesInMappings);
+                    }
+
+                    // AFTER UPDATE
+                    if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {
+                        Contact relatedContact;
+                        relatedContact = relatedContactsMap.get(affl.Contact__c);
+
+                        // ACCOUNT FIELD CHANGED
+                        if (affl.Primary__c && affl.Account__c != null && afflOld.Account__c != null
+                            && affl.Account__c != afflOld.Account__c && lookupFieldAPIName != null) {
+
+                            if (affl.Affiliation_Type__c == afflOld.Affiliation_Type__c) {
+                                //If the primary affl now points to an Account of the same type, update the
+                                //same primary field in Contact.
+                                relatedContact.put(lookupFieldAPIName, affl.Account__c);
+                            } else {
+                                //If the primary affl now points to an Account of different type, clear the old
+                                //primary field in Contact and populate the matching new field. Uncheck any other
+                                //primary affilations of same type.
+
+                                // check if there are any other Primary affiliations with same account and show an error
+                                if (idToOtherAffiliatonsWithSameAccAndCon.isEmpty() == false &&
```

Comment:
```suggestion
                                if (idToOtherAffiliatonsWithSameAccAndCon.get(affl.ID) != null &&
```
We need to use not equals null to avoid a null pointer exception in the case that the list retrieved from the map is not null before attempting to call that list's method